### PR TITLE
Initial condition input

### PR DESCRIPTION
**Pull request by tonyinme**
This is a feature to read in a netcdf file with initial conditions for velocity generated outside of amr-wind.
The feature lives within the ABL section and it is designed for it.
The number of points in the netcdf file (nx, ny, nz) needs to be the same as in the simulation.

The current implementation works when using one processor, but it doesn't work correctly with more than one processor (reads in the wrong ordering). 
This is because the netcdf reading is still missing something (need help here).
Once this feature is fixed, the PR should become ready for review.

I am tagging the group that has been helping with this development.
@michaeljbrazell @gantech @mbkuhn @rybchuk

### CODE REVIEW

**Comment by tonyinme** [comment]
This code is ready for review.  Unit tests pass.

Another PR will follow with a python script to generate the netcdf file used in the input.

**Comment by gantech** [comment]
This looks good. If you can get rid of the extra comments, that'll be good.

File: amr-wind/wind_energy/ABLFieldInit.cpp [hunk 1]
```diff
@@ -33,6 +34,9 @@ ABLFieldInit::ABLFieldInit()
 
     pp_abl.query("init_tke", m_tke_init);
 
+    // Use input from netcdf file
+    pp_abl.query("initial_condition_input_file", m_ic_input);
+
```
Comment (gantech) [inline amr-wind/wind_energy/ABLFieldInit.cpp:39]: We probly need to perform a few checks. First, that there are no mesh refinement zones enabled. Second, this option will fail or default to some baseline if netcdf is not enabled.

**Review by gantech — APPROVED** [review]

File: amr-wind/wind_energy/ABLFieldInit.cpp [hunk 2]
```diff
@@ -126,6 +130,77 @@ void ABLFieldInit::operator()(
...
+        // Set the size of the arrays to the total number of points in this
+        // processor
+        uvel2.resize(count[0] * count[1] * count[2]);
+        vvel2.resize(count[0] * count[1] * count[2]);
+        wvel2.resize(count[0] * count[1] * count[2]);
+
+        // Read the velocity components u and v
+        uvel.get(uvel2.data(), start, count);
+        vvel.get(vvel2.data(), start, count);
+        wvel.get(vvel2.data(), start, count);
```
Comment (michaeljbrazell) [inline amr-wind/wind_energy/ABLFieldInit.cpp:187]: should this be wvel2.data()?

Comment (tonyinme) [reply amr-wind/wind_energy/ABLFieldInit.cpp]: good catch! just fixed it :)

File: amr-wind/wind_energy/ABLFieldInit.cpp (hunk 1 above)
Comment (tonyinme) [reply amr-wind/wind_energy/ABLFieldInit.cpp:45]: these fixes have been added.

**Comment by tonyinme** [comment]
This PR is ready. 
Here is a cool video showing the turbulence from an ABL simulation initialized with an initial condition from a machine learning algorithm.

Thanks @mbkuhn @gantech @michaeljbrazell @rybchuk 


https://user-images.githubusercontent.com/11055361/200077352-6d3c4a43-a414-40ca-94ee-557df909dbe9.mp4

**Pull request merged**